Add language support

Maufeat opened the pull request:
Adding device language support and added the "Back" string for various languages.

fincs commented on common/language.c:
```diff
@@ -277,6 +275,22 @@ const char* const g_strings[StrId_Max][16] =
+    [StrId_Actions_Back] =
+    {
+        STR_EN("Back"),
+        STR_ES("Espalda"),
```
lol nope, this means "Back" as in body part! This should be "Atrás". Please do not use google translate; get actual translators to do the job or just leave STR_EN and the others untranslated...

yellows8 commented on common/text.c:
```diff
@@ -1,9 +1,17 @@
-
 static int s_textLang = /*CFG_LANGUAGE_EN*/1;
 
+void textInit(void) {   
```
There's zero error checking here.

Should call setExit at the end of this func too.

yellows8 commented on common/language.c:
```diff
@@ -1,7 +1,5 @@
 
-//TODO: Update this once libnx supports settings get-language.
-
 #define STR_JP(_str) [/*CFG_LANGUAGE_JP*/0] = _str
```
These defines need updated with the libnx set.h SetLanguage_* enums.

yellows8 commented on common/text.c:
```diff
@@ -1,9 +1,17 @@
 
-//TODO: Update this once libnx supports settings get-language.
-
 static int s_textLang = /*CFG_LANGUAGE_EN*/1;
```
This should also be updated with a SetLanguage enum.

plutooo commented on nx_main/main.c:
```diff
@@ -12,11 +12,12 @@ int main(int argc, char **argv)
     setsysGetColorSetId(&theme);
-    themeStartup((ThemePreset)theme);
+    themeStartup((ThemePreset)theme);    
+    textInit();       
```
Whitespace

plutooo commented on nx_main/main.c:
```diff
@@ -12,11 +12,12 @@ int main(int argc, char **argv)
-
-    ColorSetId theme;
+                                    
+    ColorSetId theme;   
```
Whitespace

plutooo commented on nx_main/main.c:
```diff
@@ -12,11 +12,12 @@ int main(int argc, char **argv)
     appletSetScreenShotPermission(1);
-
-    ColorSetId theme;
+                                    
```
Whitespace

Maufeat commented:
Added checks, call `sysExit()` at the end and using now the SetLanguage enum. And ofc the whitespaces. Sorry for those gtranslate strings, I've removed them.

yellows8 commented on common/text.c:
```diff
@@ -1,22 +1,32 @@
-    s_textLang = Language;
-}
+    Result rc = setInitialize();
+    if (R_FAILED(rc)) s_textLang = SetLanguage_ENUS;
```
Just init s_textLang here without R_FAILED(rc), and remove the duplicate lines below.

yellows8 commented:
Check for the SWITCH define & use the hard-coded lang-ID values when building for PC.

yellows8 commented on common/text.c:
```diff
@@ -6,16 +6,14 @@ void textInit(void) {
     Result rc = setInitialize();
-    if (R_FAILED(rc)) s_textLang = SetLanguage_ENUS;
+    s_textLang = SetLanguage_ENUS;
     if (R_SUCCEEDED(rc))
```
The {} aren't needed any more, merge each block into a single line.

yellows8 commented:
SetLanguage_* aren't available for PC.

Maufeat commented:
Oh ups, I hope the last commit is okay?

yellows8 commented on common/text.c:
```diff
@@ -1,15 +1,36 @@
+    s_textLang = SetLanguage_ENUS;
+    if (R_SUCCEEDED(rc)) rc = setGetSystemLanguage(&LanguageCode);
+    if (R_SUCCEEDED(rc)) rc = setMakeLanguage(LanguageCode, &Language);
+    if (R_SUCCEEDED(rc)) s_textLang = Language;
```
Change it to: ```if (R_SUCCEEDED(rc) && Language < 16) ...```

yellows8 commented on nx_main/main.c:
```diff
@@ -16,7 +16,8 @@ int main(int argc, char **argv)
     setsysGetColorSetId(&theme);
-    themeStartup((ThemePreset)theme);
+    themeStartup((ThemePreset)theme);    
+    textInit();
```
Missing from pc_main.

yellows8 commented:
Merging but have to disable it since the font doesn't support all used languages.
